plugin/pkg/proxy/connect.go

```diff
@@ -148,3 +182,22 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 }
 
 const cumulativeAvgWeight = 4
+
+// Function to return an empty response with TC (truncated) bit set.
+// This is to mitigate the effects of an upstream server sending UDP responses that exceed the maximum size of 512 bytes.
+func truncateResponse(response *dns.Msg) *dns.Msg {
+
+	truncatedResponse := response.Copy()
+
+	// Clear out Answer, Extra, and AuthenticatedData sections
+	truncatedResponse.Answer = nil
+	truncatedResponse.Extra = nil
+
+	// Clearing AD bit to indicate that the response is not authenticated.
+	truncatedResponse.AuthenticatedData = false
```

Comment:
IMO, we should not modify this bit unless you have a compelling use case you can illustrate that this fixes.

Reply:
Sure, removed resetting AD bit here.